pom.xml

```
@@ -45,8 +45,8 @@
         <jboss.snapshots.repo.id>jboss-snapshots-repository</jboss.snapshots.repo.id>
         <jboss.snapshots.repo.url>https://s01.oss.sonatype.org/content/repositories/snapshots/</jboss.snapshots.repo.url>
 
-        <quarkus.version>3.2.9.Final</quarkus.version>
-        <quarkus.build.version>3.2.9.Final</quarkus.build.version>
+        <quarkus.version>3.7.0.CR1</quarkus.version>
+        <quarkus.build.version>3.7.0.CR1</quarkus.build.version>
```

```suggestion
        <quarkus.version>3.7.0</quarkus.version>
        <quarkus.build.version>3.7.0</quarkus.build.version>
```
AFAIK, should be ready now.

---

On the versions I'm not sure about alignment. I looked for things in the quarkus runtime dependency tree that were managed and tried to update appropriately.

Now I see there is a script in set-quarkus-version, which I've updated to consider all dependency properties. This pulls in a few more alignments - but I had to also remove a couple of child pom overrides[*], one downgrade, and the osgi version change to 6.0 did not seem to work either. And of course if our property names do not line up with what is in quarkus, there will be things missed here as well. The current overlap (from the output of the script) is:

Setting 2.2.224 to h2
Setting 14.0.21.Final to infinispan
Setting 2.0.1 to jakarta.mail
Setting 6.2.7.Final to resteasy
Setting 2.0.6 to slf4j
Setting 2.2.3.Final to wildfly-elytron
Setting 4.25.1 to liquibase
Setting 6.0.0 to osgi
Setting 8.0.33 to mysql-jdbc
Setting 42.7.1 to postgresql-jdbc
Setting 3.3.2 to mariadb-jdbc
Setting 12.4.2.jre11 to mssql-jdbc
Setting 23.3.0.23.09 to oracle-jdbc
Setting 2.2 to hamcrest
Setting 4.2.0 to awaitility
Setting 1.19.3 to testcontainers

- I switched the version and dependency name  around in the pr for better readability.

Just like with the logging changes that ar proposed here, we can also consider removing some of the version overrides that we have to align better with quarkus - then re-introduce them only if we need to customize the version.

On using 3.7.0 - I didn't determine the reason, but trying that version locally doens't work for me - maven can't access the pom. 

* I'm not so sure about how something like the resteasy legacy version is currently working. It seems to miss that the resteasy spi is a transitive dependency as well.

---

With regards to the deps versions alignment. IIRC, it should not be needed to do manually, nor using the script. @Pepo48 Did some work around it to remove most of the overrides (to implicitly align with quarkus-bom) and keep only those strictly needed for a reason.

---

From a quick look, seems the explicitly stated versions are mostly for testsuites.

---

@vmuzikar yes, some of those are for tests. A breakdown of what I think this looks like:

Versions where we trail quarkus - we will spawn follow-on issues to upgrade:
- bc-fips to 1.0.2.4 
- liquibase to 4.25.1
- osgi to 6.0

Versions where we are ahead of quarkus, and that are used for more than tests - might be worth a quarkus issue for them to upgrade:
- jakarata.mail

Versions where we are trail quarkus, and that are used for more than tests, but seem to upgrade without an issue:
- infinispan
- resteasy
- slf4j
- wildfly-elytron

We can consider removing our version overrides for these - but resteasy and wildfly-elytron have some specfic overrides though that need accounted for in child poms.

The rest are used for tests, and updating to the later version associated with quarkus seems fine.

---

> We can consider removing our version overrides for these - but resteasy and wildfly-elytron have some specfic overrides though that need accounted for in child poms. We can also split these upgrade task off to separate issues if that seems better.

Yes, I'd address it as a follow-up.

Infinispan should probably not be upgraded with Quarkus as per https://github.com/keycloak/keycloak/issues/21363#issuecomment-1629117378 (CC @keycloak/store)

---

The additional version alignment will be deffered to https://github.com/keycloak/keycloak/issues/26535

---

To circle back to @mabartos 3.7.0 still does not work for me locally:

```
[ERROR]   The project org.keycloak:keycloak-parent:999.0.0-SNAPSHOT (/home/shawkins/workspaces/keycloak/keycloak/pom.xml) has 1 error
[ERROR]     Non-resolvable import POM: The following artifacts could not be resolved: io.quarkus.platform:quarkus-bom:pom:3.7.0 (absent): io.quarkus.platform:quarkus-bom:pom:3.7.0 was not found in https://maven.repository.redhat.com/techpreview/all/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of redhat-techpreview-all-repository has elapsed or updates are forced @ line 355, column 25 -> [Help 2]
```

And a similar failure happened with a gha build:

```
2024-01-26T17:39:44.2977248Z ##[error]Unable to process file command 'output' successfully.
2024-01-26T17:39:44.2980367Z ##[error]Invalid format '[ERROR] Non-resolvable import POM: Could not find artifact io.quarkus.platform:quarkus-bom:pom:3.7.0 in central (https://repo.maven.apache.org/maven2) @ line 355, column 25'
```

cc @geoand

---

Until the platform is released (next week), you need to use `io.quarkus` as the group id

---

Thanks @geoand I'll make the update once that happens.

---

@vmuzikar as far as I can tell the basic follow-up task of alignment would look like https://github.com/shawkins/keycloak/compare/iss26150...shawkins:keycloak:iss26535 - from there we would still need to tackle those upgrades.

---

> Until the platform is released (next week), you need to use `io.quarkus` as the group id

Yeah, let's use CR1 for now until Platform is released.

> as far as I can tell the basic follow-up task of alignment would look like

Thanks! Can you please create an issue for it? Probably belongs under this epic: https://github.com/keycloak/keycloak/issues/21356

---

Comment about the override of the Infinispan version: Upgrade it to a later version if Quarkus already includes it. Still, keep the separate BOM in the root pom.xml as in the previous months we've been tracking usually a more recent version of Infinispan with the latest fixes we needed, and IMHO this will continue at least for KC24. 

If with the latest Quarkus upgrade we're upgrading to a newer version of Infinispan and h2, the explicit mention of those dependencies can vanish from the root pom.xml in this or a following PR.

---

> Thanks! Can you please create an issue for it? Probably belongs under this epic: https://github.com/keycloak/keycloak/issues/21356

Yes that commit is already targeted toward #26535

> Comment about the override of the Infinispan version: Upgrade it to a later version if Quarkus already includes it.

I mispoke about infinispan being an upgrade - it's an overlapping dependency, but we're already even with quarkus. The corrected categorization of dependency alignment is on #26535